### DISCUSSION TITLE
tutorials/i18n/localization_using_gettext

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#255] [User-contributed notes]
# tutorials/i18n/localization_using_gettext

In addition to Importing translations in CSV format, Godot also supports loading translation files written in the GNU gettext format (text-based.po and compiled.mo since Godot 4.0). Advantages: get...

https://docs.godotengine.org/en/stable/tutorials/i18n/localization_using_gettext.html

**Comment by PanHoracy** [comment]
In order to have an exported variable included in the generated pot file, the variable must be named as "text", "title", "filters", "script", or end with "_text" (like "my_var_text"). 

This is based on this issue: https://github.com/godotengine/godot/issues/90053

**Comment by Sithoid** [comment]
One important caveat the docs don't specify: gettext has a different intended workflow than CSV. Instead of using keys, the source must be in English (or other human-readable language), because that's what the translation software (like MemoQ, Poedit etc) will expect. In case you still want to have the best of both worlds (rigid source keys AND plurals) or break the workflow in some other way, here's a set of open-source tools that can handle conversions: https://docs.translatehouse.org/projects/translate-toolkit/en/latest/commands/poswap.html
